File: Selfie/Selfie/SelfieTableViewCell.m
//
//  SelfieTableViewCell.m
//  Camera
//
//  Created by Jo Albright on 10/19/15.
//  Copyright © 2015 Jo Albright. All rights reserved.
//

#import "SelfieTableViewCell.h"

@interface SelfieTableViewCell ()

@property (weak, nonatomic) IBOutlet UIImageView * selfieImageView;

@end

@implementation SelfieTableViewCell

- (void)setSelfie:(PFObject *)selfie {
    
    _selfie = selfie;
    [self updateInfo];
    
}

//- (void)prepareForReuse {
//    
//    [self updateInfo];
//    
//}

- (void)updateInfo {
    
    NSLog(@"%@",self.selfie);
    PFFile * userImageFile = self.selfie[@"image"];
    [userImageFile getDataInBackgroundWithBlock:^(NSData * imageData, NSError *error) {
        if (!error) {
            
            self.selfieImageView.image = [UIImage imageWithData:imageData];
            
        }
    }];
    
    
    
//    self.selfieImageView.image = self.selfie[@"image"];
    
}

@end

File: Selfie/Selfie/SelfieTableViewCell.swift
//
//  SelfieTableViewCell.swift
//  Selfie
//
//  Created by Jo Albright on 10/30/15.
//  Copyright © 2015 Jo Albright. All rights reserved.
//

import UIKit
import Parse

class SelfieTableViewCell: UITableViewCell {

    var selfie: PFObject? {
        
        didSet {
            
            updateInfo()
            
        }
        
    }
    
    @IBOutlet weak var selfieImageView: UIImageView!

    func updateInfo() -> Void {
        
        NSLog("%@",selfie!)
        let userImageFile = selfie?["image"] as? PFFile
        
        
        userImageFile?.getDataInBackgroundWithBlock { (imageData, error) in
            
            if error == nil {
                
                self.selfieImageView.image = UIImage(data: imageData!)
                
            }
            
        }
        
    }

}










